### ISSUE TITLE
[Download][Accessibility] Duplicate links, can't add additional information to link text 

### ISSUE THREAD

**Issue by jakublamprecht** [adobe/aem-core-wcm-components#737]
## Bug Report

**Current Behavior**
Currently the Download component has a few accessibility issues which makes it not AA compliant:
- both title and the action button link to the same destination which causes duplication and redundancy of links. Either title should not be a link, or the whole card should be treated as a link.
- there's no possibility to add additional information about the link to the action link. More often than not designs will show just a single 'download' word, but for screen reader users, the link should inform the user that this link will cause a resource to be downloaded. The user should also be informed what is being downloaded and what's the size and format of the file.

I provided a few possible approaches to this problem that were mention in WCAG guidelines in the "possible solutions" section. Which one do you think would fit the best? If the component was to be AAA compliant, seems like wrapping the whole component with a link would be the best option.

Resources:

https://www.sitepoint.com/15-rules-making-accessible-links/ - point 6., 7.
https://accessabilly.com/proposal-for-a-more-accessible-download-link/
AA: https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-refs.html
AAA: https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-link.html

**Expected behavior/code**
Download component is compliant with AA accessibility requirements.

**Environment**
- Core Components version (e.g. 2.6.0)

**Possible Solution**
For the first issue:
Removing the link from title or treating the whole "download" component as a link.

For the second issue:
- Wrapping the whole download component with a link (but beware of the length of content inside the link, since users can't skip the text of the link).
- Adding aria-label
- Adding title attribute to the link
- Adding additional text to the link which will be visually hidden, but still available for screen readers.
- using aria-describedby to provide information about filename, filesize and extension by definition list above download link

**Comment by nathanmckean** [comment]
Just noting that these same issues are present in the teaser component when you choose to add a CTA (e.g. read more) but can't add an aria-label (e.g. read more about xyz) and duplicated links when you link the image and or the title. In addition, that same component places the image in the HTML before the heading and headings are supposed to lead content (a screen reader skipping over headings would miss the image). I only note this as the solution (e.g. offering the whole card as clickable, or aria-hiding the duplication etc is the same. Great write up here: https://inclusive-components.design/cards/

**Comment by gabrielwalt** [comment]
Possible solution:
- Add aria-hidden and role-presentation to the title link for screen readers to skip the link (see for eg. https://stackoverflow.com/questions/38754694/accessibility-wcag-2-0-redundant-links-and-responsive-website-navigation)
- Add aria-label to the download link with file format and size

**Comment by vladbailescu** [comment]
Alternatively we could add a content policy option to allow removing the title link.
